Source/NSPopover.m

```diff
@@ -34,101 +34,290 @@
+                     defer: (BOOL)flag
+{
+  self = [super initWithContentRect: contentRect
+                          styleMask: aStyle
+                            backing: bufferingType
+                              defer: flag];
+  if (self)
+    {
+      [super setContentView: AUTORELEASE([[GSPopoverView alloc] initWithFrame: contentRect])];
+    }
+  return self;
+}
+
+- (void) orderOut: (id)sender
+{
+  [super orderOut: sender];
+  [self close];
+}
+
+- (BOOL) canBecomeKeyWindow 
+{
+  return NO;
+}
+
+- (BOOL) canBecomeMainWindow 
+{
+  return NO;
+}
+
+- (void) setContentView: (NSView *)view
+{
+  [[super contentView] addSubview: view];
+}
+
+- (NSView *) contentView
+{
+  NSArray *subviews = [[super contentView] subviews];
+
+  if ([subviews count] == 0)
+    {
+      return nil;
+    }
+  
+  return [subviews objectAtIndex: 0];
+}
+
+@end
 
 /* Class */
 @implementation NSPopover
 
 /* Properties */
+
+/**
+ * Sets the animate flag.  If YES then the popover will animate when it appears or disappears.
+ */
 - (void) setAnimates: (BOOL)flag
 {
   _animates = flag;
 }
 
+/**
+ * Returns current value of the animate flag.
+ */
 - (BOOL) animates
 {
   return _animates;
 }
 
+/** 
+ * Sets ths appearance of the popover.  Minimal is the default.  HUD is not supported.
+ */ 
 - (void) setAppearance: (NSPopoverAppearance)value
 {
   _appearance = value;
 }
 
+/**
+ * Returns the current appearance setting.
+ */
 - (NSPopoverAppearance) appearance
 {
   return _appearance;
 }
 
+/**
+ * Sets current popup behavior.  Valid settings are:
+ * NSPopupBehaviorApplicationDefined, NSPopupBehaviorTransient, 
+ * NSPopupBehaviorSemiTransient.
+ */
 - (void) setBehavior: (NSPopoverBehavior)value
 {
   _behavior = value;
 }
 
+/**
+ * Returns current behavior setting
+ */
 - (NSPopoverBehavior) behavior
 {
   return _behavior;
 }
 
+/**
+ * Accepts an NSSize value for the current content size.
+ */
 - (void) setContentSize: (NSSize)value
 {
   _contentSize = value;
 }
 
+/**
+ * Returns an NSSize representing the size of the NSPopover content view.
+ */
 - (NSSize) contentSize
 {
   return _contentSize;
 }
 
+/**
+ * Sets the contentViewController.  If in a storyboard this is automatically set
+ * but if this is in a model that is NOT a storyboard (nib, xib, gorm, etc) then there must be
+ * a model with the same name as the class name of the contentViewController.  Also,
+ * This model must have the "view" outlet set to the view that will be shown
+ * in the popup, if both of these conditions are not met, then the code will
+ * throw an NSInternalInconsistency exception to report the issue back to the
+ * user.
+ */
 - (void) setContentViewController: (NSViewController *)controller
 {
+  if ([NSStoryboard mainStoryboard] == nil)
+    {
+      NSString *controllerClassName = NSStringFromClass([controller class]);
+      BOOL loaded = [NSBundle loadNibNamed: controllerClassName
+				     owner: controller];
+      if (!loaded)
+	{
```

Comment:
Indentations seems wrong here.

Reply:
Hmm... weird.  I will try to do untabify on it as it looks right in emacs.

Reply:
Still looks wrong, but this might be an artefact of GitHub.

Reply:
Fixed indention by using spaces.